### PR TITLE
Changes necessary to change from alevin to alevin-fry in scxa droplet quantification workflow

### PR DESCRIPTION
**Pull request by nadjano**
Two changes are necessary to the scxa-control-workflow if we switch from alevin to alevin-fry in the scxa droplet quantification workflow: 

- Make a specific transcript_to_gene.txt that is required for alevin-fry

Currently a two-column  transcript_to_gene.txt is produced, but alevin-fry required a three-column one three the third column is the splice status of the transcript.
This transcript_to_gene file is made by running `pyroe make_splici`.

- Add AEExperimentType to config file for droplet quantification workflow

The AEExperimentType field in the idf file specifies the type of experiment. 
We pass that information to the experiment configuration file that is used in the droplet quantification workflow (https://github.com/ebi-gene-expression-group/scxa-droplet-quantification-workflow/blob/e6db214a9a75472192521832c7083153851d00d9/main.nf#L9).

We need to do this because alevin-fry quantifies spliced and unspliced transcripts for single cell and single nucleus. 

But when we read in the matrix we need to speficify wether it's **snRNA** (intronic regions are quantified) or **scRNA**  (intronic regions are not quantified). See https://github.com/COMBINE-lab/pyroe/blob/main/README.md#load_fry-notes for more information.

So basically the config file changes from this:

```
params{
    name = 'E-MTAB-7094'
    organism = 'mus_musculus'
    protocol = '10xv2'

....
```

to this:

```

params{
    name = 'E-MTAB-7094'
    organism = 'mus_musculus'
    protocol = '10xv2'
    experimentType = 'single_cell'

....
```

### CODE REVIEW

**Review by nadjano — COMMENTED** [review]
wanted to comment on a commit, not here ...

File: main.nf [hunk 1]
```diff
@@ -397,6 +397,7 @@ IS_DROPLET.into{
     IS_DROPLET_PROT
     IS_DROPLET_FOR_EXP_SPECIES_TEST
     IS_DROPLET_FOR_REUSE_QUANT
+    IS_DROPLET_FOR_T2G
```
Comment (pmb59) [inline main.nf]: Following naming conventions in the rest of the workflow, I would use 'IS_DROPLET_FOR_TRANSCRIPT_TO_GENE' . Or please add a comment indicating what T2G means

The idea is to keep a unified naming system

File: envs/atlas-gene-annotation-manipulation.yml [hunk 2]
```diff
@@ -1,3 +1,4 @@
 name: atlas-gene-annotation-manipulation
 dependencies:
   - atlas-gene-annotation-manipulation=1.1.0
+  - pyroe
```
Comment (pmb59) [inline envs/atlas-gene-annotation-manipulation.yml]: maybe add here `pyroe >= X.X.X `than then current version you are using

File: main.nf [hunk 3]
```diff
@@ -791,17 +793,24 @@ process transcript_to_gene {
...
+       
 
     output:
         set val(espTag), file('transcript_to_gene.txt') into TRANSCRIPT_TO_GENE
 
     """
-    gtf2featureAnnotation.R --gtf-file $referenceGtf --version-transcripts \
-        --parse-cdnas $referenceFasta  --parse-cdna-field "transcript_id" --feature-type \
+    if [ $isDroplet == 'True' ]; then
+        pyroe make-splici ${referenceFasta} ${referenceGtf} 90 splici_transcriptome
```
Comment (pmb59) [inline main.nf:837]: why are you using here read-length = 90?

File: bin/sdrfToNfConf.R [hunk 4]
```diff
@@ -1078,10 +1079,10 @@ configs <- lapply(species_list, function(species){
...
           uri_select <- apply(species.protocol.sdrf[,uri_cols], 2, function(x) basename(x) == files)         
                               
-          missing_uri_files <- files[which(! apply(apply(species.protocol.sdrf[,uri_cols], 2, function(x) basename(x) == files), 1, any))]
-          if (length(missing_uri_files) > 0){
-            stop(paste("Can't find URIs matching files:", paste(missing_uri_files, collapse=',')))
-          }
+          #missing_uri_files <- files[which(! apply(apply(species.protocol.sdrf[,uri_cols], 2, function(x) basename(x) == files), 1, any))]
+          #if (length(missing_uri_files) > 0){
+          #  stop(paste("Can't find URIs matching files:", paste(missing_uri_files, collapse=',')))
+          #}
```
Comment (pmb59) [inline bin/sdrfToNfConf.R]: I read your comment in the commit on why you have commented this chunk

> If the lines are included I get this error
> Command error: Error in apply(apply(species.protocol.sdrf[, uri_cols], 2, function(x) basename(x) == : dim(X) must have a positive length Calls: lapply -> FUN -> lapply -> FUN -> which -> apply Execution halted.
> I think it's related to this PR https://github.com/ebi-gene-expression-group/scxa-control-workflow/pull/34.

but we need to be able to understand what is happening here. Did you notice this happening for all droplet experiments? Is it `uri_cols` definition causing the problem? Is adding `experimentType` to params list is triggering this error?

**Review by pmb59 — CHANGES_REQUESTED** [review]
Great job @nadjano ! My major concern is the error on the sdrfToNfConf.R code chunk that you left commented.

LGTM in general, but I'll leave @pcm32 to give the final approval.

File: main.nf (hunk 3 above)
Comment (nadjano) [reply main.nf:837]: @pmb59 we also need the read length to build the splici transcriptome for the index, so I think we need to set the same read length for all experiments, because we only want to have one index per species. 
In the alevin-fry paper they write: "We also set the flank length as
the read length minus five, although the quantification results appear very robust to the specific flank length chosen (Supplementary Information)", so I think it should not be a problem to set one read length for all experiments. But we maybe should set it to 150.

**Review by pmb59 — APPROVED** [review]

**Review by pmb59 — APPROVED** [review]
